src/sequences/SequenceCached.ts

```diff
@@ -16,77 +19,78 @@ import { SequenceClassDefault } from './SequenceClassDefault';
 export class SequenceCached extends SequenceClassDefault {
     name = 'Cached Base';
     description = 'A base class for cached sequences';
-    entries = 0;
-    protected cache: number[];
-    protected newSize = 1;
+    protected cache: bigint[] = [];
+    protected lastCached: number;
+    protected cachingTo: number;
+    protected blocksize: number;
 
     /**
      *Creates an instance of SequenceCached.
      * @param {*} ID the ID of the sequence
-     * @param {number} max_entries specifies the maximum number of entries. Defaults to 0, which means no maximum, if not given.
+     * @param {number} first specifies the smallest valid index of the series; defaults to 0
+     * @param {number} last specifies the largest valid index of the series; defaults to Infinity.
+     * @param {number} blocksize specifies how many values to put in cache at each additonal fill; defaults to 128.
      */
-    constructor (ID: number, max_entries?: number) {
+    constructor (ID: number,
+                 first?: number, last?: number, blocksize?:number) {
         super(ID);
-	if (max_entries !== undefined) {
-            this.entries = max_entries;
-	}
-        this.cache = [];
+        this.first = first ?? 0;
+        this.last = last ?? Infinity;
+        this.blocksize = blocksize ?? 128;
+        this.lastCached = this.first - 1;
+        this.cachingTo = this.lastCached;
     }
 
     initialize(): void {
         if (this.ready) return;
-        if (this.isValid) {
-            if (this.entries > 0) {
-                this.newSize = this.entries;
-		if (this.newSize > 100) this.newSize = 100;
-            } else {
-                this.newSize = 100;
-            }
-            this.fillCache();
-            this.ready = true;
-            return;
-        }
-
-        throw "Sequence is not valid. Run validate and address any errors."
+        super.initialize();
+        this.ready = false;
+        this.cachingTo = min(this.lastCached + this.blocksize, this.last);
+        this.fillCache();
+        this.ready = true;
     }
 
     resizeCache(n: number): void {
-        this.newSize = this.cache.length * 2;
-        if (this.newSize < 100) {
-            this.newSize = 100;
-        }
-        if (n + 1 > this.newSize) {
-            this.newSize = n + 1;
-        }
-	if (this.entries > 0 && this.newSize > this.entries) {
-            this.newSize = this.entries;
-        }
+        this.cachingTo = max(this.lastCached + this.blocksize,
+                             2*this.lastCached, n+1)
+        if (this.cachingTo > this.last) this.cachingTo = this.last;
```

Comment:
Do you mean you like to avoid one-line `if (cond) doAthing();` in favor of
```
if (cond) {
  doAthing();
}
```
Again, I think we have different philosophies here. In my view, in code formatting vertical space is the most scarce resource. Especially as my vision has become poorer, there's only so many lines that I can get on screen, and seeing the pattern of several lines can really help in understanding code. So for something like a small, simple error check or trivial adjustment to a value based on a short condition, I think using up three lines of vertical space can be a big waste.

Then I agree there could be some confusion about where conditionals start/end, etc. So to avoid that confusion I follow the guideline that one-liners never have braces, and that if there have to be two statements inside, then of course I need braces and always split it into multiple lines; I never put the opening and closing brace of a conditional on the same line.

So if we can compromise that one-liner 'if's with no else and just one statement after the conditional all on one line are OK in moderation, and that all other conditionals (with else, with multiple statements, or that don't fit on one line) use the brace-and-line-break style, that would be great. If that's really tough on your code-reading sensibilities, then I will have to adjust to whatever standard the group adopts. For example, in this particular case I will remove the offending conditional in favor of a max().